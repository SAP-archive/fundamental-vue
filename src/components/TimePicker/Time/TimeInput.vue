<template>
  <div class="fd-time__input">
    <input
      @input="$emit('update', $event.target.value)"
      class="fd-form__control"
      maxlength="2"
      type="text"
      :value="value"
      v-bind="$attrs"
      v-on="$listeners"
    />
  </div>
</template>

<script>
export default {
  inheritAttrs: false,
  name: "FdTimeInput",
  props: ["value"]
};
</script>
