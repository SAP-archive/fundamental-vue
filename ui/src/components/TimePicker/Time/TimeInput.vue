<template>
  <div class="fd-time__input">
    <input
      @input="$emit('update', $event.target.value)"
      class="fd-form__control"
      maxlength="2"
      type="text"
      :value="value"
      v-bind="$attrs"
      v-on="$listeners"
    />
  </div>
</template>

<script lang="ts">
import Vue from "vue";

export default Vue.extend({
  inheritAttrs: false,
  name: "FdTimeInput",
  props: ["value"]
});
</script>
