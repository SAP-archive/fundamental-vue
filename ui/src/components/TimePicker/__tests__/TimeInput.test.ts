import { mount } from "@vue/test-utils";
import TimeInput from "../Time/TimeInput.vue";

describe("TimeInput", () => {
  let value = "12";
  const placeholder = "Enter value";
  const ariaLabel = "Time";
  const onInput = jest.fn();

  const wrapper = mount(TimeInput, {
    propsData: {
      id: "input",
      placeholder,
      value,
      disabled: false,
      ariaLabel: ariaLabel
    },
    listeners: {
      input: onInput
    }
  });

  const input = wrapper.find("input");
  expect(input).toBeDefined();

  it("renders correctly", () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it("renders correct input value & attributes", () => {
    verifyInputValue(value);
    expect(input.attributes("placeholder")).toBe(placeholder);
    expect(input.attributes("aria-label")).toBe(ariaLabel);
  });

  it("change and verify value", () => {
    (input.element as HTMLInputElement).value = value;
    verifyInputValue("09");
  });

  const verifyInputValue = async (value: string) => {
    input.trigger("input");

    const inputEvents = wrapper.emitted("input");
    expect(inputEvents).toBeDefined();
    const inputEventsCount = inputEvents.length;
    expect(inputEventsCount).toBeGreaterThan(0);
    expect(wrapper.emitted("input")[inputEventsCount - 1][0]).toEqual(value);
    expect((input.element as HTMLInputElement).value).toEqual(value);

    await wrapper.vm.$nextTick();
    expect(wrapper.props()).toBe(value);
  };
});
