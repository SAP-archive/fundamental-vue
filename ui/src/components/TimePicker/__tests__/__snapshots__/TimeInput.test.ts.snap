// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeInput renders correctly 1`] = `
<div
  class="fd-time__input"
>
  <input
    aria-label="Time"
    class="fd-form__control"
    id="input"
    maxlength="2"
    placeholder="Enter value"
    type="text"
  />
</div>
`;
